pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1038,7 +1075,7 @@ func (cf *changeFrontier) Next() (rowenc.EncDatumRow, *execinfrapb.ProducerMetad
 				// Detect whether this boundary should be used to kill or restart the
 				// changefeed.
 				if cf.frontier.boundaryType == jobspb.ResolvedSpan_RESTART {
-					err = changefeedbase.MarkRetryableErrorWithTimestamp(err, cf.frontier.boundaryTime)
```

Comment:
Interestingly, we don't need to pass the boundary timestamp anymore because checkpointing does it for us.